### PR TITLE
luminous: multisite: data sync error repo processing does not back off on empty

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/35980

### CODE REVIEW

**Comment by smithfarm** [comment]
Non-trivial conflict resolution in the last commit

**Comment by smithfarm** [comment]
@cbodley Any idea what to do about the build error https://github.com/ceph/ceph/pull/24318#issuecomment-425360952 ?

**Comment by cbodley** [comment]
sorry @smithfarm, i had only commented on the [mimic backport](https://github.com/ceph/ceph/pull/24319#issuecomment-425429434). the utime_t constructor that this depends on was added in https://github.com/ceph/ceph/pull/21893. could you please try pulling those commits into both of these backports?

**Comment by smithfarm** [comment]
```
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/include/encoding.h:29:0,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/include/compact_map.h:16,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/include/mempool.h:30,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/common/buffer.cc:22:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/ceph_time.h:474:43: error: ‘void_t’ in namespace ‘std’ does not name a template type
 template <typename Clock, typename = std::void_t<>>
                                           ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/ceph_time.h:474:49: error: expected ‘>’ before ‘<’ token
 template <typename Clock, typename = std::void_t<>>
                                                 ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/ceph_time.h:478:36: error: ‘void_t’ is not a member of ‘std’
 struct converts_to_timespec<Clock, std::void_t<decltype(
                                    ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/ceph_time.h:478:36: error: ‘void_t’ is not a member of ‘std’
```

File: src/common/ceph_time.h [hunk 1]
```diff
@@ -452,62 +459,30 @@ namespace ceph {
...
+inline timespan to_timespan(signedspan z) {
+  ceph_assert(z >= signedspan::zero());
+  return std::chrono::duration_cast<timespan>(z);
 }
 
-// C++ Overload Resolution requires that our encode/decode functions
-// be defined in the same namespace as the type. So we need this
-// to handle things like ::encode(std::vector<ceph::real_time // > >)
+// detects presence of Clock::to_timespec() and from_timespec()
+template <typename Clock, typename = std::void_t<>>
```
Comment (smithfarm) [inline src/common/ceph_time.h]: @cbodley I'm guessing `std::void_t` is a no-no in luminous?

**Comment by yuriw** [comment]
test this please

File: src/common/ceph_time.h (hunk 1 above)
Comment (cbodley) [reply src/common/ceph_time.h]: yeah, afraid so. i think the best way forward (backward?) now is to drop those extra commits, and just add a single utime_t overload for coarse_real_time. i've pushed a copy of this branch to https://github.com/cbodley/ceph/commits/wip-35980 with a new commit https://github.com/cbodley/ceph/commit/df29bb3d2d8ae302e8ed1227421885ba2a5aea88 that does that

**Comment by smithfarm** [comment]
@cbodley OK - done.

**Comment by yuriw** [comment]
[wip-yuri5-testing-2018-10-02-2126-luminous](https://trello.com/c/AdQmRWwv/430-wip-yuri5-testing-2018-10-02-2126-luminous)

**Pull request merged**